diff --git a/node_modules/ts698/dl698/DataUtils.js b/node_modules/ts698/dl698/DataUtils.js
index a79e0a0..7bc1b61 100644
--- a/node_modules/ts698/dl698/DataUtils.js
+++ b/node_modules/ts698/dl698/DataUtils.js
@@ -163,7 +163,7 @@ var DataUtils = /** @class */ (function () {
         var length = str.length;
         var data = buffer_1.Buffer.alloc(length / 2);
         for (var i = 0; i < length; i += 2) {
-            var firstCharacter = DataUtils.digit(str.charAt(0).charCodeAt(0), 16);
+            var firstCharacter = DataUtils.digit(str.charAt(i).charCodeAt(0), 16);
             var secondCharacter = DataUtils.digit(str.charAt(i + 1).charCodeAt(0), 16);
             if ((firstCharacter == -1) || (secondCharacter == -1)) {
                 throw new Error("string is not a legal hex string.");
